--- origsrc/ghostscript-9.19/CYGWIN-PATCHES/README	1970-01-01 01:00:00.000000000 +0100
+++ src/ghostscript-9.19/CYGWIN-PATCHES/README	2016-06-29 11:02:57.397150100 +0200
@@ -0,0 +1,174 @@
+ghostscript
+-------------------------------------------------------------------------------
+GNU Ghostscript is a Postscript interpreter capable of converting PS
+files into a number of printer output formats.  Ghostscript can also
+render PS files into a number of graphics file formats.
+
+Runtime requirements:
+  cygwin-2.5.2-1
+  libbz2_1-1.0.6-2
+  libexpat1-2.2.0-0
+  libfontconfig1-2.11.95-2
+  libfreetype6-2.5.5-2
+  libgcc1-5.4.0-1
+  libICE6-1.0.9-1
+  libiconv2-1.14-3
+  libidn11-1.29-1
+  libintl8-0.19.7-1
+  libjbig2-2.0-14
+  libjpeg8-1.4.2-1
+  liblcms2_2-2.6-1
+  liblzma5-5.2.2-1
+  libpaper1-1.1.24-2
+  libpng16-1.6.20-1
+  libSM6-1.2.2-1
+  libtiff6-4.0.6-1
+  libuuid1-2.25.2-2
+  libX11_6-1.6.3-1
+  libXau6-1.0.8-1
+  libxcb1-1.11.1-1
+  libXdmcp6-1.1.2-1
+  libXt6-1.1.5-1
+  zlib0-1.2.8-3
+  +
+  ghostscript-fonts-std-8.11-1
+  ghostscript-fonts-other-6.0-1
+  gawk-4.1.3-1
+  sed-4.2.2-3
+  coreutils-8.25-3
+  which-2.20-2
+
+Build requirements:
+  cygwin-2.5.2-1
+  cygport-0.22.0-1
+  bash-4.3.42-4
+  gcc-core-5.4.0-1
+  binutils-2.25-4
+  make-4.2.1-1
+  sed-4.2.2-3
+  gawk-4.1.3-1
+  coreutils-8.25-3
+  findutils-4.6.0-1
+  autoconf2.5-2.69-3
+  autoconf-13-1
+  automake1.15-1.15-1
+  automake-9-1
+  libtool-2.4.6-3
+  pkg-config-0.29-1
+  lndir-1.0.3-1
+  ...
+  and corresponding <XXX>-devel-<y.y.y> packages from the runtime section
+
+Canonical homepage:
+  http://ghostscript.com/
+
+Canonical download:
+   http://www.ghostscript.com/download/
+
+License:
+  see /usr/share/doc/ghostscript-<Pack-Ver>/LICENSE
+
+Language:
+  C
+
+-------------------------------------------------------------------------------
+
+Build instructions:
+  unpack ghostscript-<Pack-Ver>-<Cyg-Rel>-src.tar.xz
+    if you use setup to install this src package, it will be
+ unpacked under /usr/src automatically
+  cd /usr/src
+  cygport ghostscript.cygport all
+
+This will create:
+  /usr/src/ghostscript-<Pack-Ver>-<Cyg-Rel>-src.tar.xz
+  /usr/src/ghostscript-<Pack-Ver>-<Cyg-Rel>.tar.xz
+  /usr/src/ghostscript-debuginfo-<Pack-Ver>-<Cyg-Rel>.tar.xz
+  /usr/src/libgs<abi>-<Pack-Ver>-<Cyg-Rel>.tar.xz
+  /usr/src/libgs-devel-<Pack-Ver>-<Cyg-Rel>.tar.xz
+
+Or use 'cygport ghostscript.cygport prep' to get a patched source directory
+
+To find out the files included in the binary distribution, you can
+use "cygcheck -l ghostscript ghostscript-debuginfo libgs<abi> libgs-devel", or
+browse the listing for the appropriate version at <http://cygwin.com/packages/>.
+
+-------------------------------------------------------------------------------
+
+Port Notes:
+
+----- [64bit release] version ghostscript-9.19-1 --- Wed 29 Jun 11:01:01 2016 ---
+
+ o Update to latest upstream
+ o Build for cygwin 2.5.2 with gcc-5.4.0
+ 
+----- [64bit release] version ghostscript-9.15-2 --- Thu 12 Feb 16:38:58 2015 ---
+
+ o Build for cygwin 1.7.34 with gcc-4.9.2
+ o Build against libpng16
+ 
+----- [64bit release] version ghostscript-9.15-1 --- Sat 06 Dec 18:59:08 2014 ---
+
+ o Update to latest upstream
+ o Build for cygwin 1.7.33 with gcc-4.8.3
+ 
+----- [64bit release] version ghostscript-9.06-2 --- Mon 30 Apr 13:46:01 2013 ---
+
+ o Build for cygwin 1.7.19 with gcc-4.8.0
+ o Added libpaper support
+ 
+----- [64bit release] version ghostscript-9.06-1 --- Wed 24 Apr 10:27:01 2013 ---
+
+ o Build for cygwin 1.7.19 with gcc-4.8.0
+ o This version is missing libpaper support
+ 
+----- version ghostscript-9.06-1 --- Tue 04 Sep 14:22:03 2012 ---
+
+ o Build for cygwin 1.7.16 with gcc-4.5.3
+ o Uses cygport-0.11.0 for .hint files generation
+ o Build against libpng15
+
+----- version ghostscript-9.04-1 --- Thu 10 Nov 22:36:05 2011 ---
+
+ o Build for cygwin 1.7.9 with gcc-4.5.3
+
+----- version ghostscript-8.63-2 --- Thu 27 Nov 13:14:53 2008 ---
+
+ o Fixed missing dependency on libcairo2
+   (http://cygwin.com/ml/cygwin/2008-11/msg00531.html)
+
+----- version ghostscript-8.63-1 --- Tue 25 Nov 10:54:50 2008 ---
+
+ o Update to latest upstream
+ o gs -h displayed the wrong documentation directory
+
+----- version ghostscript-8.62-3 --- Sun 23 Nov 18:46:06 2008 ---
+
+ o Recompiled against latest X11R7.4
+ o Implemented the suggestions from Yaakow, see
+   http://cygwin.com/ml/cygwin/2008-11/msg00411.html
+
+   - Removed the native/X11 split
+     (ghostscript-base and ghostscript-x11 packages obsolete, contents now
+     in ghostscript package)
+   - Fixed font paths
+   - Added shared library and corresponding devel packages
+
+----- version ghostscript-8.62-2 --- Sun 13 Apr 15:54:28 2008 ---
+
+ o Fixed a bug in referencing the wrong ghostscript executable in the
+   ghostscript helper scripts
+ o gs -h displayed the wrong documentation directory
+
+----- version ghostscript-8.62-1 --- Sun 06 Apr 17:05:33 2008 ---
+
+ o Maintainer change
+ o Switched to cygport build framework
+ o update-alternatives is used to switch between the X11 and non-X11
+   version of gs
+
+Cygwin port maintained by: Dr. Volker Zell  <dr.volker.zell@oracle.com>
+
+For original port information see /usr/share/doc/Cygwin/ghostscript-{8.15|8.50}.README
+
+Please address all questions to the Cygwin mailing list at <cygwin@cygwin.com>
--- origsrc/ghostscript-9.19/CYGWIN-PATCHES/ghostscript-8.15.README	1970-01-01 01:00:00.000000000 +0100
+++ src/ghostscript-9.19/CYGWIN-PATCHES/ghostscript-8.15.README	2016-06-29 10:49:36.214412800 +0200
@@ -0,0 +1,71 @@
+Release Notes -- GNU Ghostscript for Cygwin
+Version 8.15, Release 2 (8.15-2)
+(23 December, 2004)
+
+Version 7.05 was maintained by Dario Alcocer
+Version 8.15 was prepared by Rajesh Balakrishnan (rajeshb@infosys.com)
+(Retaining the original notes by Dario)
+
+
+The Cygwin version of GNU Ghostscript now supports both a native Win32
+version and a X11 version for Cygwin/X.  A single source package
+is used to build both versions.  Three separate packages are used:
+
+  * A base package, ghostscript-base, which contains the support
+    files necessary for both the native and X11 versions, as well
+    as the manual pages and documentation.
+
+  * A native package, ghostscript, which contains the native version
+    of Ghostscript (gs.exe), along with the corresponding scripts.
+
+  * An X11 package, ghostscript-x11, which contains the X11 version
+    of Ghostscript (gs.exe) and scripts.
+
+Both binary versions use the libpng and zlib shared libraries.
+
+This release now links /usr/lib/binmode.o to fix the text-mode mount
+bug reported by Sven Kohler.
+
+8.15-2 fixes unixinst.mak to install missing document files.
+
+REQUIREMENTS:
+------------
+
+The libraries and headers for the libpng and zlib shared libraries
+should be installed before using Ghostscript.  The setup program
+should take care of this for you.  At a minimum, you should install
+the ghostscript-base package, and either ghostscript or ghostscript-x11.
+You can also install all three packages if you'd like the choice of
+using either binary version.
+
+Due to very specific requirements Ghostscript has relating to
+Postscript's JPEG support, the JPEG shared library cannot be
+used, and must be statically linked instead.  The source for
+the JPEG library is included in the source package, ready to
+build with Ghostscript.
+
+BUILDING:
+--------
+You'll need to fetch the source archive, ghostscript-8.15-1-src.tar.xz,
+from a Cygwin mirror site.  At a minimum, you must have all the tools
+needed to build Cygwin programs, i.e. make, gcc, and binutils.  In
+addition, the zlib and libpng shared libraries and header files must be
+included in your Cygwin installation.
+
+You'll also need to grab the following font compressed tar files:
+
+    ftp://mirror.cs.wisc.edu/pub/mirrors/ghost/gnu/fonts:
+      ghostscript-fonts-std-8.11.tar.gz  (required)
+      ghostscript-fonts-other-6.0.tar.gz  (optional, recommended)
+
+Place the font compressed tar files in /tmp; after building the packages,
+these files can be deleted.
+
+Finally, to build the packages, run the cyg-pkg script; it will build all
+three packages for you and leave them in the directory /tmp/inst-gs.
+
+The Cygwin patches are in CYGWIN-PATCHES, along with this file.
+
+--
+Dario Alcocer -- Cygwin Ghostscript Maintainer
+alcocer@helixdigital.com -- http://www.helixdigital.com
--- origsrc/ghostscript-9.19/CYGWIN-PATCHES/ghostscript-8.50.README	1970-01-01 01:00:00.000000000 +0100
+++ src/ghostscript-9.19/CYGWIN-PATCHES/ghostscript-8.50.README	2016-06-29 10:49:36.239416000 +0200
@@ -0,0 +1,75 @@
+README -- GNU Ghostscript for Cygwin
+ghostscript-8.50
+
+REQUIREMENTS:
+------------
+
+The libraries and headers for the libpng and zlib shared libraries
+should be installed before using Ghostscript.  The setup program
+should take care of this for you.  At a minimum, you should install
+the ghostscript-base package, and either ghostscript or ghostscript-x11.
+You can also install all three packages if you'd like the choice of
+using either binary version.
+
+Due to very specific requirements Ghostscript has relating to
+Postscript's JPEG support, the JPEG shared library cannot be
+used, and must be statically linked instead.  The source for
+the JPEG library is included in the source package, ready to
+build with Ghostscript.
+
+BUILDING:
+--------
+You'll need to fetch the source archive, ghostscript-8.50-1-src.tar.xz,
+from a Cygwin mirror site.  At a minimum, you must have all the tools
+needed to build Cygwin programs, i.e. make, gcc, and binutils.  In
+addition, the zlib and libpng shared libraries and header files must be
+included in your Cygwin installation.
+
+You'll also need to grab the following font compressed tar files:
+
+    ftp://mirror.cs.wisc.edu/pub/mirrors/ghost/GPL/gs850/
+      ghostscript-fonts-std-8.11.tar.gz  (required)
+      ghostscript-fonts-other-6.0.tar.gz  (optional, recommended)
+
+Place the font compressed tar files in /tmp; after building the packages,
+these files can be deleted.
+
+Finally, to build the packages, run the cyg-pkg script; it will build all
+three packages for you and leave them in the directory /tmp/inst-gs.
+
+The Cygwin patches are in CYGWIN-PATCHES, along with this file.
+
+--------------------------------------------
+Release Notes
+--------------------------------------------
+Version 8.50, Release 1 (8.50-1)
+15 February, 2006
+Prepared by  James R. Phillips
+New Upstream Release
+------------------------------------------
+Version 8.15, Release 2 (8.15-2)
+23 December, 2004
+Prepared by Rajesh Balakrishnan (rajeshb@infosys.com)
+8.15-2 fixes unixinst.mak to install missing document files.
+------------------------------------------
+Version 7.05
+Prepared by Dario Alcocer (alcocer@helixdigital.com) -- http://www.helixdigital.com
+
+The Cygwin version of GNU Ghostscript now supports both a native Win32
+version and a X11 version for Cygwin/X.  A single source package
+is used to build both versions.  Three separate packages are used:
+
+  * A base package, ghostscript-base, which contains the support
+    files necessary for both the native and X11 versions, as well
+    as the manual pages and documentation.
+
+  * A native package, ghostscript, which contains the native version
+    of Ghostscript (gs.exe), along with the corresponding scripts.
+
+  * An X11 package, ghostscript-x11, which contains the X11 version
+    of Ghostscript (gs.exe) and scripts.
+
+Both binary versions use the libpng and zlib shared libraries.
+
+This release now links /usr/lib/binmode.o to fix the text-mode mount
+bug reported by Sven Kohler.
