--- gnupg-2.0.30/common/exechelp.c.org	2016-03-31 19:45:13.000000000 +0900
+++ gnupg-2.0.30/common/exechelp.c	2017-04-26 19:25:12.568018600 +0900
@@ -67,7 +67,7 @@
    here. Using a weak symbol works but is not portable - we should
    provide a an explicit dummy pth module instead of using the
    pragma.  */
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(USE_GNU_PTH)
 #pragma weak pth_fork
 #pragma weak pth_waitpid
 #endif
