--- origsrc/emacs-24.3.94/src/.gdbinit	2014-07-28 05:28:31.000000000 -0400
+++ src/emacs-24.3.94/src/.gdbinit	2014-10-11 12:26:41.569131000 -0400
@@ -1223,14 +1223,17 @@ document xbytecode
 end
 
 # Show Lisp backtrace after normal backtrace.
-define hookpost-backtrace
-  set $bt = backtrace_top ()
-  if backtrace_p ($bt)
-    echo \n
-    echo Lisp Backtrace:\n
-    xbacktrace
-  end
-end
+# This doesn't work on Cygwin in the context of 'thread apply all bt
+# full', so we disable it.  'xbacktrace' will still work.
+
+# define hookpost-backtrace
+#   set $bt = backtrace_top ()
+#   if backtrace_p ($bt)
+#     echo \n
+#     echo Lisp Backtrace:\n
+#     xbacktrace
+#   end
+# end
 
 # Flush display (X only)
 define ff
